woocommerce/woocommerce-gateway-stripe
# add payment method on checkout form

Fixes #2367

## Changes proposed in this Pull Request:
We need to add Stripe Link Payment method.

On checkout page, if the Stripe Link is enabled when you complete the email address, a new option "1 click checkout" will be visible
<img width="309" alt="Screenshot 2022-04-15 at 12 28 30" src="https://user-images.githubusercontent.com/82826872/163554151-4246b37b-8738-4abb-af52-d756d1fb718a.png">

If the email is already enrolled, after you complete the email a new pop-up will be shown to complete the pin ( 0 0 0 0 0 0 for testing )
<img width="473" alt="Screenshot 2022-04-15 at 12 30 45" src="https://user-images.githubusercontent.com/82826872/163554327-6ade6be7-7896-4bed-bf97-9559e4f9fed3.png">
If you complete the pin, all billing checkout fields will be updated with the last checkout information (billing address, city, state, postal code, country).








## Testing instructions
- go to `Settings -> Payment Methods`
- Enable Link payment method
- create an order, go to checkout page and complete all checkout fields. Select also 1click payment under credit card fields
- finish the order
- create another order, go to checkout page and complete the email
- a pop-up should be shown to type the pin ( 0 0 0 0 0 0 for tests )
- all checkout billing fields should be completed
- the order should be completed without errors





---


**Post merge**
